runner/utils/driver/istio/istio.go

```diff
@@ -66,7 +76,39 @@ func NewIstioDriver(auth *auth.Module, c *Config) (*Istio, error) {
 	// Start the keda external scaler
 	go kedaScaler.Start()
 
-	return &Istio{auth: auth, config: c, kube: kube, istio: istio, keda: kedaClient, kedaScaler: kedaScaler}, nil
+	waitservice := make(map[string]Deployment)
+	if err := WatchDeployments(func(eventType string, availableReplicas, readyReplicas int32, projectID, deploymentID string) {
+		switch eventType {
+		case resourceAddEvent, resourceUpdateEvent:
+			if waitservice[projectID] == nil {
+				waitservice[projectID] = Deployment{
+					deploymentID: {
+						AvailableReplicas: availableReplicas,
+						ReadyReplicas:     readyReplicas,
+					},
+				}
+			} else {
+				waitservice[projectID][deploymentID] = Replicas{
+					AvailableReplicas: availableReplicas,
+					ReadyReplicas:     readyReplicas,
+				}
+			}
+		case resourceDeleteEvent:
+			_, ok := waitservice[projectID]
+			if ok {
+				_, found := waitservice[projectID][deploymentID]
+				if found {
+					delete(waitservice[projectID], deploymentID)
+				} else {
+					delete(waitservice, projectID)
+				}
```

Comment:
The else case is wrong, what if there the deployment that you got doesn't exist but there are other deployment in the same project. You are deleting all the other deployments.

You should check len of deployments variable == 0